### ISSUE TITLE
Check BTS for the NC List [$50]

### ISSUE THREAD

**Issue by He3556** [CellularPrivacy/Android-IMSI-Catcher-Detector#264]
If the "unsupported flag" (see https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/100#issuecomment-67913431) is not set, we still need to check if every BTS is providing a NC  (Neighboring Cell) list.  The NC list is used to provide information to the cell phone to select a new antenna if there is an antenna with better signal reception than the currently connected one. But the cell phone doesn't need to scan or connect to all BTSs to get this info and saves energy. 

Our code is already reading the NC list with API calls and other methods. But there might be some phones that do not support this (like many Samsungs), so there will be an empty list. If the list is empty, then we need to set the `unsupported flag=1`. In the logcat you see when the phone is polling for the information. And on the menu you see the NC list under `Cell Information`, on those phones providing it.

If the `flag = 0` and the NC list of the currently connected Cell is empty, then we have a positive detection of a suspicious BTS. Most IMSI-catchers doesn't provide an NC list or they manipulate it. (Finding manipulations will be the next task to test for.) In particular, if we find a `cell-id` (CID) that is not in the OCID DB, this check is important. Thus we should run this test immediately.

**Comment by E3V3A** [comment]
Clarified issue OP text.

**Comment by Ueland** [comment]
Hmm, this might be a bit hard for me to implement when my device always complain about no neighboring cells. Or perhaps i should add the "is supported"-detection?

Any ideas on where in code we should add  this? (It`s own class or as a part of one of the others?)

**Comment by He3556** [comment]
The updateNeighbouringCells() is in the CellTracker.java (line 302) From there you can have a look how it is working now. I think we can set the flag in this code. But where can we save it? And we should check it again from time to time, so we are absolutely sure that it is the device that is not supporting the list.

**Comment by He3556** [comment]
In CellTracker.java line 642 (and the following code) the notifications are set. We need to add another tickertest into the case "MEDIUM" 
if (NoNCL)... for example

**Comment by E3V3A** [comment]
@He3556  That last comment is rather cryptic. Can you clarify? (And link to the code using `<url>#Lnnn-NNN`.)

**Comment by E3V3A** [comment]
BTW. What is `MEDIUM` there? We should label them with the color codes we already have documented.

**Comment by He3556** [comment]
It means Medium Alert. That is in the code and i don't want change it without a clear concept. 
If you start changing it here, you had to change it on many places in the code. Than we need to test again and find all the new bugs and so on...
And it is not important for the one who will implement it.

**Comment by E3V3A** [comment]
We're not changing anything, we're just sticking to what [we already documented](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/11#issuecomment-44670204) , and which make programming more transparent.

**Comment by He3556** [comment]
i have an idea of a fast implementation of this. If a NC List is (of a GSM Tower) is present we set the  property "NC-list present" to "1". If there is a tower without NC List and the value is "1" we have a detection. I will try to implement this and test it also, asap.

**Comment by E3V3A** [comment]
This is a very good idea. But the flag as you describe it above should probably be implemented in the `DBi_measure` table. The other flag (NC list) `unsupported` flag in #100 in [this](https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/100#issuecomment-67913431) diagram, is referring to HW/SW support as described in #235.

**Comment by He3556** [comment]
Yes why not. We can add the value "NC-list present" into the DBi_measure table. But than we need a query to see if the last BTS provided a list. If we have a property it is always the same check. So less possibilities of errors. And the DBi_measure table will be growing over the time. Could be faster to use a simple check of "NC-list present" value.

**Comment by E3V3A** [comment]
Sorry, then I don't understand what you mean with "property". Where is this property stored and how many are we talking about? For me a property is `getprop`.

**Comment by He3556** [comment]
we just need one value for that. If there is another place for saving a boolean let's do it somewhere else. We don't save our configuration (preferences or flags) in the db, or did this change the last days? Many things happened, it's hard to follow the changes.

**Comment by Nordlenning** [comment]
30 USD bounty on this issue so far
https://www.bountysource.com/issues/7557508-check-bts-for-the-nc-list

**Comment by SecUpwN** [comment]
>30 USD bounty on this issue so far

Strange, I wonder why the title did not update automatically. Manually updated the thread title.

**Comment by Nordlenning** [comment]
THANK YOU !

**Comment by SecUpwN** [comment]
@pchk, our app appears to detect a massive amount of missing neighboring cells now. Would you please be so kind and see if you can improve upon your already merged work? Thank you very much!

**Comment by pchk** [comment]
@SecUpwN hi, i was offline last month. Please can you describe in details what is wrong now? Or may be thare are some opened issues with description of problems you are mentioned in the post above?
